bot/exts/moderation/infraction/_utils.py

```diff
@@ -269,3 +272,43 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    def __init__(
+        self,
+        *,
+        allowed_users: t.Sequence[int],
+        allowed_roles: t.Sequence[int],
+        timeout: t.Optional[float] = 180,
+        message: t.Optional[Message] = None
+    ) -> None:
+        super().__init__(
+            allowed_users=allowed_users,
+            allowed_roles=allowed_roles,
+            timeout=timeout,
+            message=message
+        )
+
+        self.value: t.Optional[bool] = None
+
+    async def send(self, channel: TextChannel, *, message_content: str) -> None:
+        """Send this view to the given channel with the given message content."""
+        self.message = await channel.send(message_content, view=self)
```

Comment:
Why do you need to assign the message to `self.message`? 
It doesn't seem used to me

Reply:
The superclass uses it in their `on_timeout` function: https://github.com/python-discord/bot-core/blob/main/botcore/utils/interactions.py#L65-L70
```py
    async def on_timeout(self) -> None:
        """Remove the view from ``self.message`` if set."""
        if self.message:
            with contextlib.suppress(NotFound):
                # Cover the case where this message has already been deleted by external means
                await self.message.edit(view=None)
```